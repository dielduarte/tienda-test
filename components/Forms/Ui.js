import styled from 'styled-components';

import { Row, Column } from '../Flex';

export const Label = styled.label`
  font-size: 14px;
  margin-bottom: 2px;
`;

export const Input = styled.input`
  border: 2px solid
    ${props => (props.withError ? props.theme.red : props.theme.gray)};
  border-radius: ${props => (props.withPrefix ? '0 5px 5px 0;' : '5px')};
  padding: 10px;
  height: 40px;
  flex: 2;
  font-size: 14px;
  margin-bottom: 20px;
`;

export const FormRow = styled(Row)`
  flex: 1;
`;

export const FormColumn = styled(Column)`
  flex: 1;
`;

export const InputPrefix = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.gray};
  color: ${props => props.theme.white};
  min-width: 40px;
  height: 40px;
  border-radius: 5px 0 0 5px;
  margin-bottom: 20px;
`;

export const DraftContainer = styled(FormRow)`
  div[class^='RichTextEditor__root'],
  div[class^='RichTextEditor__editor'] {
    width: 100%;
  }

  div[class^='RichTextEditor__editor'] {
    height: 300px;
  }
`;
