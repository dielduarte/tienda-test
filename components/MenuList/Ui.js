import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import { Small } from '../Text';

export const Menu = styled.nav`
  list-style-type: none;
`;

export const MenuItemText = styled(Small)`
  margin-left: 10px;
  flex: 2;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const MenuIcon = styled(FontAwesomeIcon)`
  margin-right: 1px;
`;

export const MenuItem = styled.div`
  padding: 15px 10px;
  display: flex;
  align-items: center;
  color: ${props => props.theme.white};
  transition: all 0.2s linear;

  &:hover {
    cursor: pointer;
    color: ${props => props.theme.lightBlue};
  }
`;
