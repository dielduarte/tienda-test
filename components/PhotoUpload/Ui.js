import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

export const Container = styled.div`
  padding: 30px 20px 10px 20px;
`;

export const Label = styled.label`
  font-size: 14px;
`;

export const DashedContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  border: dashed 2px ${props => props.theme.gray};
  width: calc(25% - 20px);
  height: 150px;
  margin: 20px 20px 0 0;

  input {
    visibility: hidden;
    width: 0;
  }

  img {
    width: 100%;
    height: 100%;
    object-fit: contain;
  }

  &:hover {
    cursor: ${props => (props.withHover ? 'pointer' : 'default')};
  }
`;

export const Icon = styled(FontAwesomeIcon)`
  font-size: 40px;
  color: ${props => props.theme.gray};
`;
