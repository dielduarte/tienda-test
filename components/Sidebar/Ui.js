import styled from 'styled-components';
import { Small } from '../Text';

export const SidebarContainer = styled.aside`
  width: 340px;
  height: 100vh;
  overflow: scroll;
  background-color: ${props => props.theme.darkGray};
  border-right: 2px solid ${props => props.theme.gray};
`;

export const SidebarHeader = styled.div`
  background-color: ${props => props.theme.white};
  padding: 15px;
`;

export const Owner = styled(Small)`
  color: ${props => props.theme.gray};
`;
