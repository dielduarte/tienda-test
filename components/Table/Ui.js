import styled from 'styled-components';

export const Table = styled.table`
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;

  tr:nth-child(odd) {
    background: ${props => props.theme.white};
  }

  tr:nth-child(even) {
    background: ${props => props.theme.lightGray};
  }
`;

export const TableGroup = styled.tr`
  height: 60px;
  border-bottom: 2px solid ${props => props.theme.gray};
  padding: 0 10px;
`;

export const HeaderColumn = styled.th`
  font-size: 12px;
  text-align: left;
`;

export const BodyColumn = styled.td`
  font-size: 14px;
`;

export const TableContent = styled.div`
  background: ${props => props.theme.gray};
  padding: 10px;
  display: inline-block;
`;

export const TableImg = styled.img`
  width: 15px;
  height: 15px;
  object-fit: cover;
`;
