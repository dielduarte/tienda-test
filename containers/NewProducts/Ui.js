import React, { Fragment } from 'react';
import styled from 'styled-components';
import { Formik } from 'formik';

import DefaultLayout from '../layouts/default';
import PhotoUpload from '../../components/PhotoUpload';
import Card from '../../components/Card';
import { Input, Editor } from '../../components/Forms';
import { Row, ResponsiveRow } from '../../components/Flex';
import Button from '../../components/Button';

const FixedContainer = styled.div`
  width: 300px;
  margin-right: 20px;
`;

function Ui({ initialValues, validationSchema, addProduct }) {
  return (
    <DefaultLayout>
      <Formik
        initialValues={initialValues}
        validateOnChange={false}
        validationSchema={validationSchema}
        onSubmit={values => addProduct(values)}
      >
        {({
          handleChange,
          handleSubmit,
          setFieldValue,
          resetForm,
          values,
          errors
        }) => (
          <Fragment>
            <PhotoUpload
              label="Fotos do seu produto"
              onUploadSuccess={photos => setFieldValue('photos', photos)}
            />
            <Card bordered>
              <Input
                label="Nome"
                onChange={handleChange}
                name="name"
                value={values.name}
                errors={errors.name}
              />
              <Editor
                label="Descrição"
                onChange={state => setFieldValue('description', state)}
              />
            </Card>

            <Card bordered>
              <ResponsiveRow>
                <FixedContainer>
                  <Input
                    label="Preço"
                    prefix="R$"
                    onChange={handleChange}
                    name="price"
                    value={values.price}
                    type="number"
                    errors={errors.price}
                    min="0"
                  />
                </FixedContainer>
                <FixedContainer>
                  <Input
                    label="Estoque"
                    onChange={handleChange}
                    name="stock"
                    type="number"
                    value={values.stock}
                    min="0"
                  />
                </FixedContainer>
              </ResponsiveRow>
            </Card>

            <Row>
              <Button title="Publicar produto" onClick={handleSubmit} />
              <Button
                title="Cancelar"
                bordered
                type="secondary"
                onClick={() => resetForm()}
              />
            </Row>
          </Fragment>
        )}
      </Formik>
    </DefaultLayout>
  );
}

export default Ui;
