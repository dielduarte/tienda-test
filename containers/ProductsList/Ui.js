import React, { Fragment } from 'react';
import styled from 'styled-components';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import DefaultLayout from '../layouts/default';
import Card from '../../components/Card';
import Table, {
  TableGroup,
  BodyColumn,
  TableContent,
  TableImg
} from '../../components/Table';
import Link from '../../components/Link';
import { Regular } from '../../components/Text';
import PaginateItem from '../../components/PaginateItem';

const ProductName = styled(Link)`
  margin-left: 10px;
`;

function renderRow(item, key) {
  const hasPhotos = item.photos.length;

  return (
    <TableGroup key={key}>
      <BodyColumn>
        <TableContent>
          {hasPhotos ? (
            <TableImg src={item.photos[0]} />
          ) : (
            <FontAwesomeIcon icon="camera" />
          )}
        </TableContent>
        <ProductName>{item.name}</ProductName>
      </BodyColumn>
      <BodyColumn>
        <TableContent>{item.stock ? item.stock : '∞'}</TableContent>
      </BodyColumn>
      <BodyColumn>
        <TableContent>R$ {item.price}</TableContent>
      </BodyColumn>
      <BodyColumn>
        <TableContent>+</TableContent>
      </BodyColumn>
      <BodyColumn />
      <BodyColumn>content</BodyColumn>
    </TableGroup>
  );
}

function Ui({ store, products, page, setPage }) {
  return (
    <DefaultLayout>
      <Card>
        {!products.length ? (
          <Regular>Você ainda não cadastrou nenhum produto :(</Regular>
        ) : (
          <Table
            headers={[
              'Produto',
              'Estoque',
              'Preço original',
              'Preço promocional',
              'variações',
              'Ações'
            ]}
          >
            {products[page].map((product, key) => renderRow(product, key))}
          </Table>
        )}
      </Card>
      {products.map((p, key) => (
        <PaginateItem key={key} onClick={() => setPage(key)}>
          {key + 1}
        </PaginateItem>
      ))}
    </DefaultLayout>
  );
}

export default Ui;
