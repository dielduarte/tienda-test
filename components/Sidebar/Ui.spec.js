import { shallow } from 'enzyme';
import 'jest-styled-components';

import { SidebarContainer, SidebarHeader, Owner } from './Ui';

describe('Sidebar => Ui', () => {
  it('SidebarContainer', () => {
    const wrapper = shallow(<SidebarContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('SidebarHeader', () => {
    const wrapper = shallow(<SidebarHeader />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Owner', () => {
    const wrapper = shallow(<Owner />);
    expect(wrapper).toMatchSnapshot();
  });
});
