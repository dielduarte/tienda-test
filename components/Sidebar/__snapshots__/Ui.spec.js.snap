// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar => Ui Owner 1`] = `
<styled.h3
  className=""
/>
`;

exports[`Sidebar => Ui SidebarContainer 1`] = `
.c0 {
  width: 340px;
  height: 100vh;
  overflow: scroll;
  border-right: 2px solid;
}

<aside
  className="c0"
/>
`;

exports[`Sidebar => Ui SidebarHeader 1`] = `
.c0 {
  padding: 15px;
}

<div
  className="c0"
/>
`;
